src/core/channels/presence_channel.ts

```diff
@@ -29,17 +29,10 @@ export default class PresenceChannel extends PrivateChannel {
     super.authorize(socketId, (error, authData) => {
       if (!error) {
         authData = authData as ChannelAuthorizationData;
-        if (authData.channel_data === undefined) {
-          let suffix = UrlStore.buildLogSuffix('authenticationEndpoint');
-          Logger.error(
-            `Invalid auth response for channel '${this.name}',` +
-              `expected 'channel_data' field. ${suffix}`
-          );
-          callback('Invalid auth response');
-          return;
+        if (authData.channel_data != null) {
+          var channelData = JSON.parse(authData.channel_data);
+          this.members.setMyID(channelData.user_id);
```

Comment:
Shouldn't we verify first if `user_id` exists?

Reply:
We could, but the reason is not needed is because this will set id to `null` and then the subscription will fail, because the pusher server will reject it.